sklearn/exceptions.py

```diff
@@ -85,6 +85,25 @@ class DataConversionWarning(UserWarning):
           implementation's data-type expectations;
         - passes an input whose shape can be interpreted ambiguously.
 
+    Examples
+    --------
+
+    >>> import numpy as np
+    >>> from sklearn.metrics import pairwise_distances
+    >>> X = np.random.RandomState(0).rand(5,3)
+    >>> import warnings
+    >>> from sklearn.exceptions import DataConversionWarning
+    >>> warnings.simplefilter('always', DataConversionWarning)
+    >>> with warnings.catch_warnings(record=True) as w:
+    ...     try:
+    ...             # Will trigger warning as metric requires X.dtype=bool
+    ...             D = pairwise_distances(X, metric="jaccard")
```

Comment:
May we use `column_or_1d` instead as an example that issues a `DataConversionWarning` warning? I think users would run into that situation more often.